Sources/StreamChatUI/Utils/ViewPaginationHandling/ViewPaginationHandling.swift

```diff
@@ -11,3 +11,14 @@ protocol ViewPaginationHandling {
     var onNewTopPage: (() -> Void)? { get set }
     var onNewBottomPage: (() -> Void)? { get set }
 }
+
+typealias StatefulViewPaginationHandlingBlock = (((Int) -> Void, @escaping (Error?) -> Void) -> Void)
+protocol StatefulViewPaginationHandling {
```

Comment:
Is this protocol really necessary? We don't seem to have tests for it (it would be great if we do btw) and this doesn't seem something that will have variations.

Reply:
Which kind of tests would you add for the protocol? Tests for its concrete implementation are there

Reply:
The reason i'm not sure about the protocol's purpose, is that in the way it's setup now, you can swap out the StatefulViewPaginationHandler inside InvertedScrollViewPaginationHandler and replace it with an instance of another InvertedScrollViewPaginationHandler. It seems to me that the StatefulViewPaginationHandler is more of a store for the ViewPaginationHandlers rather than a ViewPaginationHandler itself.